src/python/CRABInterface/RESTBaseAPI.py

```diff
@@ -96,9 +96,25 @@ def _initLogger(self, logfile, loglevel, keptDays=0):
         logger = logging.getLogger('CRABLogger')
         if loglevel:
             hdlr = logging.handlers.TimedRotatingFileHandler(logfile, when='D', interval=1, backupCount=keptDays)
-            formatter = logging.Formatter('%(asctime)s:%(levelname)s:%(module)s:%(message)s')
+            formatter = logging.Formatter('%(asctime)s:%(trace_id)s:%(levelname)s:%(module)s:%(message)s')
             hdlr.setFormatter(formatter)
+            # add trace_id to log with filter class
+            f = TraceIDFilter()
+            hdlr.addFilter(f)
+
             logger.addHandler(hdlr)
             logger.setLevel(loglevel)
         else:
             logger.addHandler( NullHandler() )
+
+
+class TraceIDFilter(logging.Filter):
+    """
+    Add trace_id to log record and use it in formatter.
+    """
+    def filter(self, record):
+        try:
+            record.trace_id = cherrypy.request.db['handle']['trace'].replace('RESTSQL:','')
```

Comment:
Following up a quick private discussion and for future memory (because i am sure I will forget about this), thanks @novicecpp for making sure that the `cherrypy.request.db` object is (or at least should) always be available as a "global" variable when we log from `RESTBaseApi`.
Ok, not really global because it is not shared across the whole process, but only in the specific thread that is logging, but still, we have access to that variable in this scope.